includes/widgets/rating.php

```diff
@@ -0,0 +1,161 @@
+<?php
+namespace Elementor;
+
+use Elementor\Icons_Manager;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Elementor (new) rating widget.
+ *
+ * @since 3.17.0
+ */
+class Widget_Rating extends Widget_Base {
+
+	public function get_name() {
+		return 'rating';
+	}
+
+	public function get_title() {
+		return esc_html__( 'Rating', 'elementor' );
+	}
+
+	public function get_icon() {
+		return 'eicon-rating';
+	}
+
+	public function get_keywords() {
+		return [ 'star', 'rating', 'review', 'score', 'scale' ];
+	}
+
+	protected function register_controls() {
+		$this->start_controls_section(
+			'section_rating',
+			[
+				'label' => esc_html__( 'Rating', 'elementor' ),
+			]
+		);
+
+		$this->add_control(
+			'rating_maximum_value',
+			[
+				'label' => esc_html__( 'Rating Scale', 'elementor' ),
+				'type' => Controls_Manager::SLIDER,
+				'range' => [
+					'px' => [
+						'min' => 1,
+						'max' => 10,
+					],
+				],
+				'step' => 1,
+				'default' => [
+					'size' => '5',
+				],
+			]
+		);
+
+		$this->add_control(
+			'rating_value',
+			[
+				'label' => esc_html__( 'Rating', 'elementor' ),
+				'type' => Controls_Manager::NUMBER,
+				'min' => 0,
+				'step' => 0.01,
+				'dynamic' => [
+					'active' => true,
+				],
+			]
+		);
+
+		$this->add_control(
+			'rating_icon',
+			[
+				'label' => esc_html__( 'Icon', 'elementor' ),
+				'type' => Controls_Manager::ICONS,
+				'fa4compatibility' => 'icon',
+				'skin' => 'inline',
+				'label_block' => false,
+				'skin_settings' => [
+					'inline' => [
+						'icon' => [
+							'icon' => 'eicon-star',
+						],
+					],
+				],
+				'default' => [
+					'value' => 'eicon-star',
+					'library' => 'eicons',
+				],
+			]
+		);
+
+		$this->end_controls_section();
+	}
+
+	protected function get_rating_value(): float {
+		return floatval( $this->get_settings_for_display( 'rating_value' ) );
+	}
+
+	protected function get_rating_maximum_value(): int {
+		return intval( $this->get_settings_for_display( 'rating_maximum_value' )['size'] );
+	}
+
+	protected function get_star_marked_width( $star_index ): string {
+		$rating_value = $this->get_rating_value();
+
+		if ( $rating_value >= $star_index ) {
+			return '100%';
+		} else if ( intval( ceil( $rating_value ) ) === $star_index ) {
+			return ( $rating_value - ( $star_index - 1 ) ) * 100 . '%';
+		}
+
+		return '0%';
+	}
```

Comment:
I am curious about Louis' idea about this. 

Generally speaking, I prefer to use return as quickly as possible, so that we don't have to run through the whole function unnecessarily. 

What do you think @louiswol94